src/chainstate/stacks/boot/pox_2_tests.rs

```diff
@@ -0,0 +1,423 @@
+
+            assert_eq!(
+                (reward_addrs[1].0).version,
+                AddressHashMode::SerializeP2PKH.to_version_testnet()
+            );
+            assert_eq!((reward_addrs[1].0).bytes, key_to_stacks_addr(&alice).bytes);
+            assert_eq!(reward_addrs[1].1, 512 * POX_THRESHOLD_STEPS_USTX);
+        }
+    };
+
+    // first tenure is empty
+    peer.tenure_with_txs(&[], &mut coinbase_nonce);
+
+    let alice_balance = get_balance(&mut peer, &key_to_stacks_addr(&alice).into());
+    assert_eq!(alice_balance, 1024 * POX_THRESHOLD_STEPS_USTX);
+
+    let alice_account = get_account(&mut peer, &key_to_stacks_addr(&alice).into());
+    assert_eq!(
+        alice_account.stx_balance.amount_unlocked(),
+        1024 * POX_THRESHOLD_STEPS_USTX
+    );
+    assert_eq!(alice_account.stx_balance.amount_locked(), 0);
+    assert_eq!(alice_account.stx_balance.unlock_height(), 0);
+
+    // next tenure include Alice's lockup
+    let tip =
+        SortitionDB::get_canonical_burn_chain_tip(&peer.sortdb.as_ref().unwrap().conn()).unwrap();
+    let alice_lockup = make_pox_lockup(
+        &alice,
+        0,
+        1024 * POX_THRESHOLD_STEPS_USTX,
+        AddressHashMode::SerializeP2PKH,
+        key_to_stacks_addr(&alice).bytes,
+        4,
+        tip.block_height,
+    );
+    let tip_index_block = peer.tenure_with_txs(&[alice_lockup], &mut coinbase_nonce);
+
+    // check the stacking minimum
+    let total_liquid_ustx = get_liquid_ustx(&mut peer);
+    let min_ustx = with_sortdb(&mut peer, |ref mut chainstate, ref sortdb| {
+        chainstate.get_stacking_minimum(sortdb, &tip_index_block)
+    })
+    .unwrap();
+    assert_eq!(min_ustx, total_liquid_ustx / 480);
+
+    // no reward addresses
+    let reward_addrs = with_sortdb(&mut peer, |ref mut chainstate, ref sortdb| {
+        get_reward_addresses_with_par_tip(chainstate, &burnchain, sortdb, &tip_index_block)
+    })
+    .unwrap();
+    assert_eq!(reward_addrs.len(), 0);
+
+    // check the first reward cycle when Alice's tokens get stacked
+    let tip_burn_block_height = get_par_burn_block_height(peer.chainstate(), &tip_index_block);
+    let alice_reward_cycle = 1 + burnchain
+        .block_height_to_reward_cycle(tip_burn_block_height)
+        .unwrap() as u128;
+
+    assert_eq!(alice_reward_cycle, EXPECTED_ALICE_REWARD_CYCLE);
+
+    // alice locked, so balance should be 0
+    let alice_balance = get_balance(&mut peer, &key_to_stacks_addr(&alice).into());
+    assert_eq!(alice_balance, 0);
+
+    // produce blocks until immediately before the epoch switch (7 more blocks to block height 35)
+
+    for _i in 0..7 {
+        peer.tenure_with_txs(&[], &mut coinbase_nonce);
+
+        // alice is still locked, balance should be 0
+        let alice_balance = get_balance(&mut peer, &key_to_stacks_addr(&alice).into());
+        assert_eq!(alice_balance, 0);
+    }
+
+    // now, try to use PoX 2 contract. this should _fail_.
```

Comment:
I'm not sure which of the asserts shows that this "failed".. is there one? If so, can a stronger connection be made with comments, e.g., does "failing" mean Alice has a balance of 0?